### PR TITLE
[Catalog] Fix issue related to uncaught problems during a COMMIT

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes https://github.com/duckdblabs/duckdb-internal/issues/2759

The correctness of dependencies were not being verified in the COMMIT phase.

### CODE REVIEW

File: src/catalog/dependency_manager.cpp [hunk 1]
```diff
@@ -433,17 +433,75 @@ string DependencyManager::CollectDependents(CatalogTransaction transaction, cata
...
-		// Don't do anything for this
 		return;
 	}
-
 	auto info = GetLookupProperties(object);
-	// Check if there are any entries that block the DROP because they still depend on the object
-	catalog_entry_set_t to_drop;
+	ScanDependents(transaction, info, [&](DependencyEntry &dep) {
+		auto dep_committed_at = dep.timestamp.load();
+		if (dep_committed_at > start_time) {
```
Comment (Mytherin) [inline src/catalog/dependency_manager.cpp:475]: Do we need the `start_time` here? We should not have any dependencies at this point anymore, no?

File: src/catalog/catalog_set.cpp [hunk 2]
```diff
@@ -479,14 +510,20 @@ bool CatalogSet::UseTimestamp(CatalogTransaction transaction, transaction_t time
 	return false;
 }
 
-CatalogEntry &CatalogSet::GetEntryForTransaction(CatalogTransaction transaction, CatalogEntry &current) {
+CatalogEntry &CatalogSet::GetEntryForTransaction(CatalogTransaction transaction, CatalogEntry &current, bool *visible) {
```
Comment (Mytherin) [inline src/catalog/catalog_set.cpp:513]: Can we use `bool &visible` here?

We can create a wrapper method to make this optional, e.g.:

```cpp

CatalogEntry &CatalogSet::GetEntryForTransaction(CatalogTransaction transaction, CatalogEntry &current) {
    bool visible;
    return GetEntryForTransaction(transaction, current, visible);
}

```

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good - two minor comments

File: src/catalog/dependency_manager.cpp (hunk 1 above)
Comment (Tishj) [reply src/catalog/dependency_manager.cpp:475]: The drop of the dependents has not committed yet, so we need this check
I have added clarifying comments for this

**Comment by Mytherin** [comment]
I've moved this to feature - could you resolve the merge conflicts?

File: src/include/duckdb/transaction/commit_state.hpp [hunk 3]
```diff
@@ -24,7 +24,7 @@ struct UpdateInfo;
 
 class CommitState {
 public:
-	explicit CommitState(DuckTransaction &transaction, transaction_t commit_id);
+	explicit CommitState(DuckTransaction &transaction, transaction_t commit_id, transaction_t start_time);
```
Comment (Mytherin) [inline src/include/duckdb/transaction/commit_state.hpp:27]: Minor conflict note - we can get the start_time from the DuckTransaction here

**Review by Mytherin — COMMENTED** [review]
Thanks for the fixes - one minor note, otherwise I think this is good to go after CI passes

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**